### PR TITLE
Fix chat history implementation

### PR DESCRIPTION
**Pull request by areejfnaqvi**
## Description

This PR corresponds to the following [Task](https://www.notion.so/conradlabshq/Fix-Chat-History-121059a36d584070be8c0aebd1bb46ef?pvs=4)

A user can now view chat history from previous chats and continue the conversation from where they left off

### CODE REVIEW

File: app/(chat)/chat/[id]/page.tsx [hunk 1]
```diff
@@ -50,20 +53,26 @@ export default async function ChatPage({ params }: ChatPageProps) {
...
-      role: "user"
-    }]
+    const { id, title } = chat as UserChatMessage
+    const updatedChats: ChatMessage[] = chat.messages
+      ? chat.messages
+      : [
+          {
+            id: id,
+            message: title,
+            role: 'user'
```
Comment (AwaisKamran) [inline app/(chat)/chat/[id]/page.tsx:63]: Any string used as an error/success message should be a constant or an enum

File: components/chat.tsx [hunk 2]
```diff
@@ -11,33 +11,55 @@ import { Message, Session } from '@/lib/types'
...
   const [input, setInput] = useState('')
   const messages = useSelector((state: any) => state.chat.messages)
+  const dispatch = useDispatch()
   const [mockups, setMockups] = useState(null)
   const [isCarouselOpen, setIsCarouselOpen] = useState(false)
 
   const [_, setNewChatId] = useLocalStorage('newChatId', id)
 
+  useEffect(() => {
+    if (messages.length !== 1) {
```
Comment (AwaisKamran) [inline components/chat.tsx:50]: This is to check of the message list is greater than 1? right

Comment (areejfnaqvi) [reply components/chat.tsx]: yes, updated

File: app/(chat)/chat/[id]/page.tsx (hunk 1 above)
Comment (areejfnaqvi) [reply app/(chat)/chat/[id]/page.tsx]: done

**Review by AwaisKamran — APPROVED** [review]

**Pull request merged**